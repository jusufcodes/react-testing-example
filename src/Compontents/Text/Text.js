import React from "react";

export const Text = (props) => {
  return props.children ? <p>{props.children}</p> : null;
};