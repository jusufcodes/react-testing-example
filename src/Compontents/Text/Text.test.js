import React from 'react';
import { shallow } from 'enzyme';
import renderer from "react-test-renderer";
import {Text} from "./Text";

describe("Card text component", () => {
  const demoText = "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt";

  test("Text without text", () => {
    const content = <Text />;

    expect(shallow(content).isEmptyRender()).toEqual(true);

    const contentSnapshot = renderer.create(content).toJSON();
    expect(contentSnapshot).toMatchSnapshot();
  });

  test("Text with text", () => {
    const content = <Text>{demoText}</Text>;

    expect(shallow(content).text()).toEqual(demoText);

    const contentSnapshot = renderer.create(content).toJSON();
    expect(contentSnapshot).toMatchSnapshot();
  });
});