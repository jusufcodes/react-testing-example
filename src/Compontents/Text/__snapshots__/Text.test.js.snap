// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card text component Text with text 1`] = `
<p>
  Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt
</p>
`;

exports[`Card text component Text without text 1`] = `null`;
